### PR TITLE
Add a new constructor to RigidMotion and use it in the interface

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: astronomy/solar_system_dynamics_test.cpp [hunk 1]
```diff
@@ -216,8 +217,8 @@ class SolarSystemDynamicsTest : public ::testing::Test {
     }
     RigidMotion<ICRS, ParentEquator> const to_parent_equator(
         {ICRS::origin, ParentEquator::origin, rotation->Forget()},
-        /*angular_velocity_of_to_frame=*/{},
-        /*velocity_of_to_frame_origin=*/{});
+        AngularVelocity<ICRS>{},
+        Velocity<ICRS>{});
```
Comment (eggrobin) [inline astronomy/solar_system_dynamics_test.cpp]: Keep the names, it's hepful to see that it's the velocity *of* to frame, not just that it's expressed *in* from frame.

File: ksp_plugin/interface.cpp [hunk 2]
```diff
@@ -899,7 +899,7 @@ void principia__InsertOrKeepLoadedPart(
       vessel_guid,
       main_body_index,
       FromQP<DegreesOfFreedom<World>>(main_body_world_degrees_of_freedom),
-      part_rigid_motion_inverse.Inverse(),
+      part_rigid_motion_inverse,
```
Comment (eggrobin) [inline ksp_plugin/interface.cpp]: Drop the `_inverse`. Just `part_rigid_motion`. It's cleaner.

File: physics/rigid_motion_body.hpp [hunk 3]
```diff
@@ -20,6 +20,18 @@ RigidMotion<FromFrame, ToFrame>::RigidMotion(
...
 
+template<typename FromFrame, typename ToFrame>
+RigidMotion<FromFrame, ToFrame>::RigidMotion(
+    RigidTransformation<FromFrame, ToFrame> const& rigid_transformation,
+    AngularVelocity<ToFrame> const& angular_velocity_of_from_frame,
+    Velocity<ToFrame> const& velocity_of_from_frame_origin)
+    : rigid_transformation_(rigid_transformation),
+      angular_velocity_of_to_frame_(
+          rigid_transformation_.linear_map().Inverse()(
+              angular_velocity_of_from_frame)),
```
Comment (eggrobin) [inline physics/rigid_motion_body.hpp]: Shouldn't there be a sign here? Can you add a test?

File: physics/rigid_motion_test.cpp [hunk 4]
```diff
@@ -173,19 +173,12 @@ TEST_F(RigidMotionTest, GroupoidInverse) {
...
   EXPECT_THAT(
-      d1.position() - Terrestrial::origin,
-      AlmostEquals(degrees_of_freedom_.position() - Terrestrial::origin, 0));
-  EXPECT_THAT(d1.velocity(), AlmostEquals(degrees_of_freedom_.velocity(), 4));
-  EXPECT_THAT(
-      d2.position() - Terrestrial::origin,
+      d.position() - Terrestrial::origin,
       AlmostEquals(degrees_of_freedom_.position() - Terrestrial::origin, 4));
-  EXPECT_THAT(d2.velocity(), AlmostEquals(degrees_of_freedom_.velocity(), 6));
+  EXPECT_THAT(d.velocity(), AlmostEquals(degrees_of_freedom_.velocity(), 6));
```
Comment (eggrobin) [inline physics/rigid_motion_test.cpp]: Keep `RigidMotion<F, F>`, default-constructible only (the identity), and keep this test.

File: physics/rigid_motion.hpp [hunk 5]
```diff
@@ -36,13 +36,12 @@ using quantities::si::Radian;
 template<typename FromFrame, typename ToFrame>
 class RigidMotion final {
  public:
-  static_assert(!std::is_same_v<FromFrame, ToFrame>,
-                "FromFrame and ToFrame must be different");
-
   RigidMotion(
```
Comment (eggrobin) [inline physics/rigid_motion.hpp]: Same enable_if on this one (and probably a typename = void on the other one for disambiguation).

File: physics/rigid_motion_test.cpp [hunk 6]
```diff
@@ -188,6 +188,32 @@ TEST_F(RigidMotionTest, GroupoidInverse) {
   EXPECT_THAT(d2.velocity(), AlmostEquals(degrees_of_freedom_.velocity(), 6));
 }
 
+TEST_F(RigidMotionTest, SecondConstructor) {
+  DegreesOfFreedom<Selenocentric> const earth_dof_in_selenocentric =
+      geocentric_to_selenocentric_({Geocentric::origin,
+                                    Velocity<Geocentric>{}});
+  AngularVelocity<Selenocentric> const earth_rotation_in_selenocentric;
```
Comment (eggrobin) [inline physics/rigid_motion_test.cpp]: `geocentric_to_selenocentric_` has vanishing angular velocity, so we are not testing this sign.

Involve terrestrial or lunar (or both: terrestrial_to_lunar, see above).

File: physics/rigid_motion.hpp (hunk 5 above)
Comment (pleroy) [reply physics/rigid_motion.hpp]: No can do.

**Pull request merged**